plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -380,6 +380,14 @@ void state_history_plugin_impl::plugin_shutdown() {
    accepted_block_connection.reset();
    block_start_connection.reset();
    thread_pool.stop();
+
```

Comment:
With the connections torn down, is there any risk pumping the main thread below might apply a block that ship would then not see?

Reply:
Good call! It does seem like this could push a block into the controller that would be missed by the controller.

Reply:
Actually, no I don't think that is not possible. Everything that uses `app().executor().post` when they execute they are only pushed into the priority_queue to execute. So probably should clear that out again here.

Reply:
I wonder if we even need those connections to be manually torn down

Reply:
Probably not, but that can be cleaned up when we work the real solution.